### ISSUE TITLE
allow passing workflow parameters as a file (separate from reana.yml)

### ISSUE THREAD

**Issue by lukasheinrich** [reanahub/reana-client#492]
via `-p`/`--parameters` I can do e.g.

```
reana-client run --name recast-pmsssm-1234 --file reana-demo-atlas-recast/reana.yaml -parameter foo=bar
```

but sometimes workflows have nested parameters / many parameters and listing them individually might be difficult. Being able to pass them via a file (keeping the rest of reana.yml fixed) would be good

```
reana-client run --name recast-pmsssm-1234 --file reana-demo-atlas-recast/reana.yaml --parameter-file pars.json
```

is that possible?

**Comment by mvidalgarcia** [comment]
CWL supports setting an input file in the reana.yaml. E.g. [reana-demo-cdci-crab-pulsar-integral-verification](https://github.com/reanahub/reana-demo-cdci-crab-pulsar-integral-verification/blob/2674d484cd2cb1ac8c00068e6bc53804be31b999/reana.yaml#L4). Would that fulfil your need? I guess we can easily support it for Yadage workflows [here](https://github.com/reanahub/reana-client/blob/03460b791804e287063a7b4e56cb6bf7c7d78e19/reana_client/utils.py#L158-L162).

**Comment by lukasheinrich** [comment]
is that a parameter names `input` with value `inputs.yaml ` i.e. equivalent to `-p inputs=inputs.yaml ` or JSON `{"inputs":inputs.yaml` or does this have special semanticcs within CWL?

**Comment by mvidalgarcia** [comment]
I think it must be set inside the reana.yaml at the moment (so no `-p inputs=inputs.yaml` yet), and as far as I can see it's only supported for CWL workflows.

**Comment by lukasheinrich** [comment]
it seems like this could be a source of confusion (e.g. I would have def. expected `-p inputs=inputs.yaml` equivalence) .. would adding a `parameter_file` entry in reana.yml or similar be feasible?

**Comment by mvidalgarcia** [comment]
> it seems like this could be a source of confusion (e.g. I would have def. expected -p inputs=inputs.yaml equivalence) .. 

I agree.. it looks like this feature was asked for CWL workflows and remained there. In addition, the translation of parameters from the yaml file to the workflow spec is happening at a creation time, and one can only pass runtime parameters `-p` at start time. So probably that why this equivalence you mention does not exist...

Said that, in my opinion, it makes total sense to add a parameter file option to `reana-client start`, and consequently to `reana-client run`.
